src/lib/server/orders.ts

```diff
@@ -371,6 +371,38 @@ export async function onOrderPaymentFailed(
 		},
 		{ returnDocument: 'after', session: opts?.session }
 	);
+	for (const item of order.items) {
+		if (item.freeQuantity) {
+			const freeProductSubscriptions = await collections.paidSubscriptions
+				.find({
+					[`freeProductsById.${item.product._id}`]: { $exists: true },
+					[`freeProductsById.${item.product._id}.available`]: { $gt: 0 },
+					paidUntil: { $gt: new Date() }
+				})
+				.toArray();
+
+			for (const sub of freeProductSubscriptions) {
+				await collections.paidSubscriptions.updateOne(
+					{ _id: sub._id },
+					{
+						$inc: {
+							...(item.freeQuantity && {
+								[`freeProductsById.${item.product._id}.used`]: -item.quantity
+							}),
+							...(item.freeQuantity && {
+								[`freeProductsById.${item.product._id}.available`]: +item.quantity
+							})
+						},
+						$set: {
+							updatedAt: new Date(),
+							notifications: []
+						},
+						$unset: { cancelledAt: 1 }
+					}
+				);
+			}
+		}
+	}
```

Comment:
this should only happen if the order is fully cancelled.

Like in this section: https://github.com/be-BOP-io-SA/be-BOP/pull/1847/files#diff-68e2d47d872b8e27c8f8a5ae4ccdb5409fe481e6c03a49b3c7f323530c1107c0R401-R417

```ts
	if (
		order.status !== ret.value.status &&
		(ret.value.status === 'canceled' ||
			ret.value.status === 'expired' ||
			ret.value.status === 'failed')
	) {
		await collections.scheduleEvents.updateMany(
			{
				orderId: order._id
			},
			{
				$set: {
					status: 'canceled'
				}
			}
		);
	}
```